rick-and-morty/src/App.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import './App.css';
+import {BrowserRouter as Router, Route, Switch, Link} from 'react-router-dom';
+import CharacterList from './CharacterList';
+import CharacterPage from './CharacterPage';
+import logo from './logo.png';
+
+
+function character(props) {
```

Comment:
I would probable recommend refactoring this so there is no functional component and only the CharacterPage, since there seems to be some unnecessary nesting. But the key point of passing params through the url is there, nice work!

Reply:
So instead of making the character component, i could just do everything in CharacterPage?

Reply:
Correct, you would have to change Character page to access id from this.props.match.params.id
and on line 36 you would change the returned component in the route to be CharacterPage.